### PR TITLE
Refactoring to firebase v9 modular

### PR DESCRIPTION
**Pull request by amilosmanli**
Refactored code to firebase v9 modular

- Fixes issues with filtering in lazy mode
- Fixes issues with total item count in lazy mode

This is probably a breaking change

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -1,4 +1,11 @@
...
-  "tabWidth": 2
-}
+  "trailingComma": "none",
+  "tabWidth": 2,
+  "bracketSpacing": true,
+  "arrowParens": "avoid",
+  "importOrder": ["^react(.*)$", "^firebase/(.*)$", "^@(.*)/(.*)$", "^[./]"],
+  "importOrderSeparation": true,
+  "importOrderSortSpecifiers": true
+}
```
Comment (benwinding) [inline .prettierrc:11]: This are all subjective formatting changes that make this PR way harder to review, as there's thousands of lines of linting changes 😭 

Please make a separate PR for prettier and linting changes.

**Review by benwinding — CHANGES_REQUESTED** [review]
Thanks for the contribution, but we need to remove the linting changes from this PR, as there's thousands of lines of linting code mixed into this package upgrade too

Just change it to the original `.prettierrc` file and `yarn lint` then commit that to this PR

File: .prettierrc (hunk 1 above)
Comment (amilosmanli) [reply .prettierrc:11]: Yeah, sorry about this big change. 

Though linting options and code doesn't match. For example, both .prettierrc and tslint.json states that singleQuotes should be used, but code is using double quotes. I was trying to force the linting config and added some extra configuration. 

I will open another PR with linting changes
